### DISCUSSION TITLE
Timers - Not Reliable

### DISCUSSION THREAD

**Discussion by Sc00by121** [arendst/Tasmota#17476] [Q&A] [answered]
I have around 55 Sonoff Dual R2 units running Tasmota 12.3.1

The units control streetlights and heating circuits.  The streetlight timers work on a simple sunrise/sunset and work fine however the units controlling heating circuits are not behaving as expected.

An example found today against 9 of the units;

Timer;
1 - ON at 6am
2  - OFF at 9am
3 - OFF at 1pm
4 - OFF at 6pm

The circuit came on at 6am but when I logged into the unit at 1030am it was still ON.  Looking back at syslog it goes off at 6pm.  

The other units are doing the same albeit at different times but slot 2 seems to be ignored.

Is this normal operation with the way this has been set up, is the best way to leave them ON/OFF respectively and check the ENABLED box instead?

Can someone advise on how to investigate this further?

**Comment by barbudor** [comment]
Can you give more details?
Like the outputs of commands Timer1 to Timer4
And logs with Weblog 3 at offending hours when the switching is supposed to occur

**Comment by Sc00by121** [comment]
I have set syslog and weblog to 3 and will see what comes through.

I've had syslog set to 2 for quite some time but it doesn't show anything happening at the trigger time.  How do I get the outputs of timer1 to 4?

It may be a week or so before I can get any data back from this due to the festive season

**Reply by sfromis** [reply]
Type the commands in the console. You can also combine into one command like:
`Backlog Timer1;Timer2;Timer3;Timer4`

You can also print all timers with one command, just `Timers` giving a long string.

**Comment by Sc00by121** [comment]
I'm back onto this again as they are still not behaving as they should. However I may have stumbled across a possible reason. Some units are in low Wifi coverage area and I have found that if they disconnect for any length of time it appears they reboot however when they come back online the clock has reset until they can pick up Wifi again so for the duration of being offline they have the wrong time. Trying to catch this in the logs is difficult. Is it possible to alter this behaviour?

Timers output of one unit didn't come on at 06:30 this morning;
{"Timers":"ON","Timer1":{"Enable":1,"Mode":0,"Time":"06:30","Window":0,"Days":"1111111","Repeat":1,"Output":1,"Action":1},"Timer2":{"Enable":1,"Mode":0,"Time":"09:30","Window":0,"Days":"1111111","Repeat":1,"Output":1,"Action":0},"Timer3":{"Enable":1,"Mode":0,"Time":"12:30","Window":0,"Days":"1111111","Repeat":1,"Output":1,"Action":1},"Timer4":{"Enable":1,"Mode":0,"Time":"14:30","Window":0,"Days":"1111111","Repeat":1,"Output":1,"Action":0},"Timer5":{"Enable":1,"Mode":0,"Time":"16:30","Window":0,"Days":"1111111","Repeat":1,"Output":1,"Action":1},"Timer6":{"Enable":1,"Mode":0,"Time":"21:30","Window":0,"Days":"1111111","Repeat":1,"Output":1,"Action":0},"Timer7":{"Enable":1,"Mode":0,"Time":"16:30","Window":0,"Days":"1111111","Repeat":1,"Output":2,"Action":1},"Timer8":{"Enable":1,"Mode":0,"Time":"22:30","Window":0,"Days":"1111111","Repeat":1,"Output":2,"Action":0},"Timer9":{"Enable":0,"Mode":0,"Time":"00:00","Window":0,"Days":"0000000","Repeat":0,"Output":1,"Action":0},"Timer10":{"Enable":0,"Mode":0,"Time":"00:00","Window":0,"Days":"0000000","Repeat":0,"Output":1,"Action":0},"Timer11":{"Enable":0,"Mode":0,"Time":"00:00","Window":0,"Days":"0000000","Repeat":0,"Output":1,"Action":0},"Timer12":{"Enable":0,"Mode":0,"Time":"00:00","Window":0,"Days":"0000000","Repeat":0,"Output":1,"Action":0},"Timer13":{"Enable":0,"Mode":0,"Time":"00:00","Window":0,"Days":"0000000","Repeat":0,"Output":1,"Action":0},"Timer14":{"Enable":0,"Mode":0,"Time":"00:00","Window":0,"Days":"0000000","Repeat":0,"Output":1,"Action":0},"Timer15":{"Enable":0,"Mode":0,"Time":"00:00","Window":0,"Days":"0000000","Repeat":0,"Output":1,"Action":0},"Timer16":{"Enable":0,"Mode":0,"Time":"00:00","Window":0,"Days":"0000000","Repeat":0,"Output":1,"Action":0}}

Interesting that this post (https://github.com/arendst/Tasmota/discussions/16011)  covers this potential issue, Wifi loss, restart and no NTP but no resolution

I have also followed https://github.com/arendst/Tasmota/issues/1034 and the output of my devices is;

`{"WifiConfig":{"4":"Retry"}}`

**Reply by barbudor** [reply]
WifiConfig 4 expressly instruct tasmota to reboot. This is useful only when having an alternate access point
If not, just change to 5 (wait without rebooting)

**Comment by Sc00by121** [comment]
In that case the documentation needs updating as it states it doesn't reboot when set to 4.  5 locks to an AP which is not ideal if you want it to jump to a stronger closer unit unless this has changed and the documentation has not updated for other options?

`4 = retry other AP without rebooting (default)
5 = wait until selected AP is available again without rebooting`

https://tasmota.github.io/docs/Commands/#wi-fi

**Comment by Jason2866** [comment, marked as answer]
The chance is high when not solving bad WiFi that the devices do not what the should do.
Tasmota does need WiFi connection all the time. The only real solution is to enhance the WiFi network.
And wificonfig 4 does not reboot WHEN one of the configured AP is reachable. If none is reachable it can happen device is rebooting.
wificonfig 5 is optimal for switching to strongest AP. Used with SetOption56 and SetOption57
The usual setup in a WLAN with more APS, is that all AP sends out the SAME SSID.
AP2 is more ment for a recovery scenario or when device is used at two different places.

**Reply by barbudor** [reply]
My mistake, I answered from memory which was obviously bad

So the reason for your reboot is different
Youlqy have a rule?

**Comment by Sc00by121** [comment]
So no real fix at a Tasmota level? - Boosting coverage is the only option?

What do you mean by having a rule?

**Reply by Jason2866** [reply]
Tasmota depends on WiFi. That's a must have. Like fixing a car to run without fuel.
